src/devices/cpu/i960/i960.cpp

```diff
@@ -1455,6 +1455,78 @@ void i960_cpu_device::execute_op(uint32_t opcode)
 				}
 				break;
 
+			case 0x2:   // daddc
+				/*
+					Adds bits 0 through 3 of src2 and src1 and bit 1 of the condition code (used
+					here: as a carry bit). The result is stored in bits 0 through 3 of dst. If the
+					addition results in a carry, bit 1 of the condition code is set. Bits 4 through
+					31 of src are copied to dst unchanged.
+
+					This instruction is intended to be used iteratively to add binary-coded decimal
+					(BCD) values in which the least-significant four bits of the operands
+					represent the decimal numbers 0 to 9. The instruction asssumes that the least
+					significant 4 bits of both operands are valid BCD numbers. If these bits are
+					not valid BCD numbers, the resulting value in dst is unpredictable.
+				*/
+				{
+					uint64_t res;
+
+					m_icount -= 2;
+					t1 = get_1_ri(opcode) & 0x0000000f;
+					t2 = get_2_ri(opcode) & 0x0000000f;
+					res = t2+(t1+((m_AC>>1)&1));
+					set_ri(opcode, res&0xffffffff);
+
+					m_AC &= ~0x2;   // clear C
+					// set carry
+					m_AC |= ((res) & (((uint64_t)1) << 32)) ? 0x2 : 0;
+				}
+				break;
+
+			case 0x3:   // dsubc
+				/*
+					Subtracts bits 0 through 3 of src2 and srcl and bit I of the condition code
+					(used here as a carry bit). The result is stored in bits 0 through 3 of dst. If
+					the subtraction results in a carry, bit 1 of the condition code is set. Bits 4
+					through 31 of src are copied to dst unchanged. 
+					
+					This instruction is intended to be used iteratively to add binary-coded decimal
+					(BCD) values in which the least-significant four bits of the operands
+					represent the decimal numbers 0 to 9. The instruction asssumes that the least
+					significant 4 bits of both operands are valid BCD numbers. If these bits are
+					not valid BCD numbers, the resulting value in dst is unpredictable.
+				*/
+				{
+					uint64_t res;
+
+					m_icount -= 2;
+					t1 = get_1_ri(opcode) & 0x0000000f;
+					t2 = get_2_ri(opcode) & 0x0000000f;
+					res = t2-(t1+((m_AC>>1)&1));
+					set_ri(opcode, res&0xffffffff);
+
+					m_AC &= ~0x2;   // clear C
+					// set carry
+					m_AC |= ((res) & (((uint64_t)1) << 32)) ? 0x2 : 0;
```

Comment:
This seems to suffer from similar poorly thought-out implementation to the addition function.

Reply:
Almost all of the code you've highlighted was not originally written by me. As I mentioned, I copied the existing code for the addc/subc opcodes and modified where necessary (which really just amounted to making the operands 4 bit and removing the V flag clear).

Considering daddc/dsubc do not seem to be used anywhere (at least not in any Model 2 disasms I've been working with) and the point of this PR is primarily to add dmovt, I will gladly excise the daddc/dsubc portions.

Reply:
> Almost all of the code you've highlighted was not originally written by me.  As I mentioned, I copied the existing code for the addc/subc opcodes and modified where necessary (which really just amounted to making the operands 4 bit and removing the V flag clear).

You haven’t modified where necessary to actually match the description though:
* The `addc` and `subc` instructions actually do 32-bit addition/subtraction, so taking the carry/borrow flag from the thirty-second bit makes sense in that case.
* You describe the instructions as only affecting the four least significant bits of the destination but you haven’t changed the code to actually do that.

Reply:
Thank you. Points taken and will be considered when/if I revisit this. For now, I have removed the daddc/dsubc code as I don't have the time to work on this further. PR updated.